deps/v8/src/api/api-wasm.cc

```diff
@@ -0,0 +1,317 @@
+// Copyright 2019 the V8 project authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "src/api/api-wasm.h"
+#include "src/base/memory.h"
+#include "src/common/globals.h"
+#include "src/api/api-inl.h"
+#include "src/objects/contexts.h"
+#include "src/objects/lookup.h"
+#include "src/execution/isolate.h"
+
+#define IGNORE(X) ((void) (X))
+
+namespace b = v8::base;
+namespace i = v8::internal;
+
+namespace v8 {
+namespace wasm {
+
+Memory::Memory(size_t pages, uint8_t* data) : pages_(pages), data_(data) {}
+Memory::Memory(const Memory& memory)
+    : pages_(memory.pages_), data_(memory.data_) {}
+size_t Memory::size() { return pages_ * PAGE_SIZE; }
+size_t Memory::pages() { return pages_; }
+uint8_t* Memory::data() { return data_; }
+
+Table::Table(void* tableObject) {
+  tableObject_ = (void*) new i::WasmTableObject(
+      *(i::WasmTableObject*) tableObject);
+}
+Table::Table(const Table& table) {
+  tableObject_ = (void*) new i::WasmTableObject(
+      *(i::WasmTableObject*) table.tableObject_);
+}
+Table::~Table() { delete tableObject_; }
```

Comment:
I am not sure this delete works correctly (without relying on UB). The actual object you allocated was via `new i::WasmTableObject`, but the type of `tableObject_` is `void *`.

Reply:
Great point, totally forgot about that